src/QSymbolicsBase/predefined.jl

```diff
@@ -102,22 +105,53 @@ isexpr(::OperatorEmbedding) = true
 @withmetadata struct XGate <: AbstractSingleQubitGate end
 eigvecs(g::XGate) = [X1,X2]
 symbollabel(::XGate) = "X"
+ishermitian(::XGate) = true
+isunitary(::XGate) = true
+iscommutative(::XGate) = false
```

Comment:
"commutative" is not a property of a single linear operator. Only `const*I` commutes with all other operators for a general group. I do not think this can be consistently defined.

Reply:
My bad, silly blunder of mine. I'll remove this property.